### PR TITLE
Partial data displays on frontend 

### PR DESCRIPTION
**Pull request by lin8231**
This PR makes it so that if one servlet fails the whole app doesn't work. The areas where that servlet's data is used displays an error. 
* Created component ErrorMessage. This is a super simple component where it displays "Failed to load x" where x is the child prop passed in. Used in...
  * Recent buys 
  * Long sgonks list
  * Short sgonks list 
  * Trending searches
* Also edited SelectCompetition so that if the competitionsList doesn't return properly, it shows an error message instead of going bork
* Also fixed a bug where select competition was throwing an error because a method name change that happened in a previous PR
# ~~This branch is branched off fe-refactor which has its own PR. Only the last 10 commits belong to this PR~~ rebased

### CODE REVIEW

**Review by Phoebe-Khokgawe — APPROVED** [review]

**Review by Phoebe-Khokgawe — APPROVED** [review]
LGTM :)

**Review by leonnemets — APPROVED** [review]
Nice

**Review by texm — APPROVED** [review]
Nice work

File: project/frontend/src/App.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ export const NO_COMPETITION = 0;
 function App() {
   const [user, setUser] = useState({ signedIn: false });
   const [authStateReceived, setAuthStateReceived] = useState(false);
-  const [compId, setCompId] = useState(0);
+  const [compId, setCompId] = useState(NO_COMPETITION);
```
Comment (texm) [inline project/frontend/src/App.js:17]: this is very helpful

**Pull request merged**